src/Cache/RedisFactory.php

```diff
@@ -73,6 +73,13 @@ private function normalizeServer(string $server): array
             $parsedServer['password'] = urldecode($pass);
         }
 
+        $database = $parsedServer['path'] ?? null;
+        unset($parsedServer['path']);
+
+        if ($database !== null) {
+            $parsedServer['database'] = (int) trim($database, '/');
```

Comment:
If databases are always represented by a number, I would validate that instead of casting to `int`. Otherwise, if someone provides an invalid value by mistake, the connection will fail and the error won't be very useful.

Here, if database is not null, but it is also not a number, I would throw an error indicating it.

Extracting the whole conditional block to a private method is probably desirable, to avoid nesting conditions.

Reply:
I wasn't sure if you'd want an exception thrown in case someone already had an incorrect configuration here that was currently being ignored and would become an exception but I'll change it to throw an InvalidArgumentException instead, thanks for the feedback.

Reply:
> I wasn't sure if you'd want an exception thrown in case someone already had an incorrect configuration here that was currently being ignored and would become an exception

That's actually a good point. Let's keep it then, and just add a TODO comment to change it for the next major version.